### PR TITLE
[Iteration 1]Merge spring 24 main to snp-cloud main

### PR DESCRIPTION
**Pull request by warrenlet**
**Checklist**

- [x] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x] My branch is up-to-date with the upstream `master` branch.
- [x] The tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works(if appropriate).
- [x] I have added necessary documentation (if appropriate).

**Short description of what this resolves/which [issues](https://github.com/openSUSE/osem/issues) does this fix?:**

https://github.com/snap-cloud/snapcon/pull/344

**Changes proposed in this pull request:**

- changes attribute 'submission instructions' to 'submission template' for event_types model. Changed corresponding view, controller, model, and testing code.
- Makes it such that no user (admins, non-admins) can see these toggles. An admin is still able to change these settings by going to administration->users->edit. The button for admin has also been updated from a checkbox to a toggle switch.
- Update the  INSTALL_SNAPCON.md to provide instruction to the situation of fork safty issues from MacOS

### CODE REVIEW

File: app/mailers/mailbot.rb [hunk 1]
```diff
@@ -32,11 +32,28 @@ def ticket_confirmation_mail(ticket_purchase)
...
+    if @ticket_purchase.ticket_id == YTLF_TICKET_ID
+      template_name = 'young_thinkers_ticket_confirmation_template'
+      mail(subject: "#{@conference.title} | Ticket Confirmation and PDF!", template_name: template_name)
+    end
 
-    mail(subject:       "#{@conference.title} | Ticket Confirmation and PDF!",
-         template_name: template_name)
+    # if email subject is empty, use custom template
+    if @ticket_purchase.ticket.email_subject.empty? && !@ticket_purchase.ticket.email_body.empty?
+      @ticket_purchase.ticket.email_body = @ticket_purchase.generate_confirmation_mail(@ticket_purchase.ticket.email_body)
```
Comment (fp456) [inline app/mailers/mailbot.rb:42]: The conditional statements are very self explanatory so I think this comment can be deleted

File: app/views/mailbot/custom_ticket_confirmation_template.html.erb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<%= render partial: "layouts/mailbot_header" %>
+  <div id="content">
+    <span style="white-space: pre-line">
+        <%= @ticket_purchase.ticket.email_body %>
+    </span>
+  </div>
+<%= render partial: "layouts/mailbot_footer" %>
```
Comment (cycomachead) [inline app/views/mailbot/custom_ticket_confirmation_template.html.erb:7]: Should have a newline.

```suggestion
<%= render partial: "layouts/mailbot_footer" %>

```

**Review by fp456 — CHANGES_REQUESTED** [review]
Hi! Good overall, just some clean up to do and data to be updated about the members of the team

File: app/services/email_template_parser.rb [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    values.each do |key, value|
+      if value.is_a?(Date)
+        text = text.gsub "{#{key}}", value.strftime('%Y-%m-%d') if text.present?
+      else
+        text = text.gsub "{#{key}}", value unless text.blank? || value.blank?
+      end
+    end
+    text
+  end
+  # rubocop:enable Metrics/AbcSize, Metrics/ParameterLists
```
Comment (fp456) [inline app/services/email_template_parser.rb:75]: You can remove this line

File: config/routes.rb [hunk 4]
```diff
@@ -1,3 +1,4 @@
+# rubocop:disable Metrics/BlockLength
```
Comment (fp456) [inline config/routes.rb:1]: If this is a comment, and the line won't be used, you can delete

File: db/migrate/20230418211400_create_currency_conversions.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class CreateCurrencyConversions < ActiveRecord::Migration[7.0]
+  def change
+    create_table :currency_conversions do |t|
+      t.decimal :rate
+      t.string :from_currency
+      t.string :to_currency
+      t.references :conference, null: false, foreign_key: true
+
```
Comment (fp456) [inline db/migrate/20230418211400_create_currency_conversions.rb:8]: you can remove this empty line

File: spec/features/currency_conversions_spec.rb [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    end
+
+    after do
+      sign_out
+    end
+
+    it 'add a currency conversion', feature: true do
+      visit admin_conference_currency_conversions_path(conference.short_title)
+      click_link 'Add Currency Conversion'
+
```
Comment (fp456) [inline spec/features/currency_conversions_spec.rb:21]: you can remove this empty line

File: spec/features/currency_conversions_spec.rb [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      fill_in 'currency_conversion_from_currency', with: 'USD'
+      fill_in 'currency_conversion_to_currency', with: 'EUR'
+      fill_in 'currency_conversion_rate', with: '0.89'
+
```
Comment (fp456) [inline spec/features/currency_conversions_spec.rb:25]: you can remove this empty line

**Review by fp456 — CHANGES_REQUESTED** [review]
just some empty lines, tiny stuff but again overall looks good :)

**Review by cycomachead — APPROVED** [review]
Really minor things, but I will merge and might make a tidy-ing PR myself then deploy soon.

**Pull request merged**